src/Raven.Server/Documents/Queries/QueryMetadata.cs

```diff
@@ -412,6 +413,15 @@ private void Build(BlittableJsonReaderObject parameters)
                 }
             }
 
+            if (TrafficWatchManager.HasRegisteredClients)
```

Comment:
you cannot set it here, QueryMetadata is being cached. You need to handle that in 'SetupTimings' method IMO

Reply:
Done.